python/oneseismic/scan/scan.py

```diff
@@ -264,15 +263,23 @@ def scan(stream, action):
     stream.seek(textheader_size, io.SEEK_CUR)
     chunk = stream.read(binary_size)
     binary = segyio.field.Field(buf = chunk, kind = 'binary')
-    skip = action.binary_header(binary)
-    if skip > 0:
-        stream.seek(skip * textheader_size, io.SEEK_CUR)
+
+    # Information needed by the scan program
+    intp    = parseint(endian)
```

Comment:
I think endian should be parsed to an int in __main__. Parsing this to an int is really command line argument parsing.

Reply:
You mean to create the parseint instance at callsite and pass that, rather than initialising it from the endian string here? In that case, should we also remove the parseint instance from all scanners? And pass it to each function like so
```
for scanner in scanners:
    scanner.scan_binary_header(binary, intp)
```

Thus avoiding the need to explicitly pass `endian` to each scanner on initialization

Reply:
Never mind, I was a bit quick to comment here without properly looking into it.